### PR TITLE
Wrong schema configuration

### PR DESCRIPTION
**Pull request by pgs-gmartowski**

### CODE REVIEW

File: src/core/config/schema/ConfigSchema.ts [hunk 1]
```diff
@@ -52,15 +52,20 @@ export const ConfigSchema: Joi.JoiObject = Joi.object().keys({
     expirationDate: Joi.string().required(),
     notificationFrame: Joi.string().required(),
     securityCode: Joi.string().required()
-  }),
-  components: ComponentsConfigSchema,
+  }).default({}),
+  components:  Joi.object().keys({
+    defaultPaymentType: Joi.string().allow(''),
+    requestTypes: Joi.array().items(Joi.string().valid('THREEDQUERY', 'AUTH')),
```
Comment (james-simpson-ST) [inline src/core/config/schema/ConfigSchema.ts]: Shouldn't this allow all of these options? 'ACCOUNTCHECK', 'AUTH', 'JSINIT', 'RISKDEC', 'SUBSCRIPTION', 'THREEDQUERY'

So it should be 
```javascript
requestTypes: Joi.array().items(
    Joi.string().valid('ACCOUNTCHECK', 'AUTH', 'JSINIT', 'RISKDEC', 'SUBSCRIPTION', 'THREEDQUERY')
  )
```

Comment (pgs-gmartowski) [reply src/core/config/schema/ConfigSchema.ts]: true, I forgot that there are couple of places where it has to be added explicitly

Comment (pgs-gmartowski) [reply src/core/config/schema/ConfigSchema.ts]: done

File: src/core/config/schema/ConfigSchema.ts [hunk 2]
```diff
@@ -52,15 +52,20 @@ export const ConfigSchema: Joi.JoiObject = Joi.object().keys({
...
     notificationFrame: Joi.string().required(),
     securityCode: Joi.string().required()
-  }),
-  components: ComponentsConfigSchema,
+  }).default({}),
+  components:  Joi.object().keys({
+    defaultPaymentType: Joi.string().allow(''),
+    requestTypes: Joi.array().items(Joi.string().valid('ACCOUNTCHECK', 'AUTH', 'JSINIT', 'RISKDEC', 'SUBSCRIPTION', 'THREEDQUERY')),
+    paymentTypes: Joi.array().items(Joi.string().allow('')),
+    startOnLoad: Joi.boolean().allow('').allow('')
```
Comment (wirus15) [inline src/core/config/schema/ConfigSchema.ts]:
```
allow('').allow('')
```

btw, I think `allow()` is not necessary for the boolean. Plus, what exactly `allow('')` does (with empty string)?

Comment (wirus15) [reply src/core/config/schema/ConfigSchema.ts]: If it supposed to mean that this option can be empty, I think I saw a `.optional()` in the docs

Comment (pgs-gmartowski) [reply src/core/config/schema/ConfigSchema.ts]: oops, yes there is too much allow() ;)

Comment (pgs-gmartowski) [reply src/core/config/schema/ConfigSchema.ts]: @wirus15 not exactly, here is discussion in Joi's repo - https://github.com/hapijs/joi/issues/482 - seems like it's really odd solution to make a property `not required`

**Pull request merged**